src/ansys/pre_commit_hooks/tech_review.py

```diff
@@ -0,0 +1,734 @@
+        print("The pyproject.toml and setup.py files do not exist")
+        print("Cannot get the author and maintainer name and email, project name, and version\n")
+
+    # Returns True if file is complaint or False if it is not compliant,
+    # the name of the project from the configuration file, and the type
+    # of configuration file found in the repository
+    return is_compliant, project_name, config_file
+
+
+def check_pyproject_toml(
+    repo_path: str,
+    author_maint_name: str,
+    author_maint_email: str,
+    is_compliant: bool,
+    non_compliant_name: bool,
+) -> tuple:
+    """
+    Check pyproject.toml file for correct naming convention, version, author, and maintainer.
+
+    Parameters
+    ----------
+    repo_path: str
+        Path of the repository being checked.
+    author_maint_name: str
+        Project author and maintainer's name.
+    author_maint_email: str
+        Project author and maintainer's email.
+    is_compliant: bool
+        ``True`` if the repository is compliant.
+        ``False`` if the repository is not compliant.
+    non_compliant_name: bool
+        ``True`` if the repository's name is not in the form ansys-*-* and it is permitted.
+        ``False`` if the repository's name is in the form ansys-*-*.
+
+    Returns
+    -------
+    bool
+        ``True`` if the pyproject.toml file's information was correct.
+        ``False`` if the pyproject.toml file had missing or incorrect information.
+    str
+        Name of the project from the pyproject.toml file.
+    """
+    name = ""
+    # Load pyproject.toml
+    with open(repo_path / "pyproject.toml", "r") as project_file:
+        config = toml.load(project_file)
+        project = config.get("project")
+
+        # Check the project name follows naming conventions: ansys-{product}-{library}
+        # Ignore this check if non_compliant_name argument is True
+        if not non_compliant_name:
+            name = project.get("name", "DNE")
+            if (name == "DNE") or (
+                (name != "DNE") and not bool(re.match(r"^ansys-[a-z]+-[a-z]+$", name))
+            ):
+                is_compliant = False
+                print("Project name does not follow naming conventions")
+
+        # Check the project version follows Semantic versioning
+        project_version = project.get("version", "DNE")
+        if project_version != "DNE":
+            try:
+                version = semver.Version.parse(project_version)
+            except ValueError:
+                is_compliant = False
+                print("Project version does not follow semantic versioning")
+
+        # Check the project author and maintainer names and emails match argument input
+        category, metadata = ["authors", "maintainers"], ["name", "email"]
+        # [('authors', 'name'), ('authors', 'email'), ('maintainers', 'name'),
+        # ('maintainers', 'email')]
+        output = list(product(category, metadata))
+
+        for combo in output:
```

Comment:
Use pattern matching in here and avoid the usage of array-like indices. This enhances readability:

```suggestion
        for key, value in output:
```